From 67fcc3f8c34cf1fc7983b1acc4752eb9e2cfe769 Mon Sep 17 00:00:00 2001
From: Ruoyu Zhong <zhongruoyu@outlook.com>
Date: Thu, 9 Mar 2023 17:54:12 +0800
Subject: [PATCH] Fix CMake install location

`GNUInstallDirs` needs to be included before `CMAKE_INSTALL_INCLUDEDIR`
can be used.

Signed-off-by: Ruoyu Zhong <zhongruoyu@outlook.com>
---
 CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 75a80e28..97829743 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,6 +39,8 @@ configure_file(
 
 include_directories_ispc(${CMAKE_CURRENT_BINARY_DIR}/openvkl/include/${PROJECT_NAME})
 
+include(GNUInstallDirs)
+
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/openvkl/include/${PROJECT_NAME}/version.h
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}
 )
@@ -48,7 +50,6 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR}/openvkl/include/${PROJECT_NAME}/versio
 option(BUILD_EXAMPLES "Build example applications." ON)
 option(BUILD_BENCHMARKS "Build benchmark applications." OFF)
 
-include(GNUInstallDirs)
 include(CTest) # Adds 'BUILD_TESTING' variable
 
 if (BUILD_TESTING)
